infrastructure/src/main/java/org/corfudb/infrastructure/log/FileChannelPerf.java

```diff
@@ -0,0 +1,183 @@
+package org.corfudb.infrastructure.log;
+
+
+import com.codahale.metrics.Timer;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.ServerContext;
+import org.corfudb.util.MetricsUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
```

Comment:
I think its better to just extend FileChannel and override whichever methods you want to intercept.

Reply:
FileChannel is an abstract class and some functions are not overidable, such as isOpen, isClose. It makes things complicate.

Reply:
I wasn't precise enough, I meant `FileChannelImpl` then you can just override the methods that you are interested in.

Reply:
The constructor for FileChannelImpl is very complicate and use some system values. It is not feasible to extend that class. the FileChannel.open function return a FileChannelImp.